core/src/test/kotlin/io/specmatic/core/ScenarioTest.kt

```diff
@@ -658,4 +662,254 @@ class ScenarioTest {
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = JSONArrayValue(listOf(StringValue("test1"), StringValue("test2")))
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactlyInAnyOrder("[0]{string}", "[1]{string}")
+        }
+
+        @Test
+        fun `calculatePath should return empty set for unsupported body pattern types`() {
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(
+                    body = StringPattern()
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = StringValue("test")
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).isEmpty()
+        }
+
+        @Test
+        fun `calculatePath should handle nested complex patterns`() {
+            val nestedPattern = JSONObjectPattern(
+                mapOf(
+                    "items" to ListPattern(
+                        JSONObjectPattern(
+                            mapOf("value" to AnyPattern(listOf(StringPattern(), NumberPattern()))),
+                            typeAlias = "(Item)"
+                        )
+                    )
+                ),
+                typeAlias = "(Container)"
+            )
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(body = nestedPattern),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = JSONObjectValue(mapOf(
+                    "items" to JSONArrayValue(listOf(
+                        JSONObjectValue(mapOf("value" to StringValue("test"))),
+                        JSONObjectValue(mapOf("value" to NumberValue(42)))
+                    ))
+                ))
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactlyInAnyOrder(
+                "{Container}.items[0].{Item}.value{string}",
```

Comment:
This assertion should be:

            assertThat(paths).containsExactlyInAnyOrder(
                "{Container}.items[0]{Item}.value{string}",
                "{Container}.items[1]{Item}.value{number}"
            )